Source/PINDiskCache.m

```diff
@@ -991,6 +1005,8 @@ - (void)setObject:(id <NSCoding>)object forKey:(NSString *)key fileURL:(NSURL **
                 [self->_dates setObject:date forKey:key];
             }
             
+            [_knownKeys addObject:key];
```

Comment:
There's a lot of collection access in this method - both reading from the two dictionaries and writing to all three collections. This path would benefit the most from combining the structures into one dictionary.